packages/main/src/SliderBase.js

```diff
@@ -0,0 +1,659 @@
+
+		// Draw the tickmarks as a patern over the gradient background
+		const tickmarksGradientdPattern = `0 center / calc((100% - ${tickmarkWidth}) / (${this._tickmarksAmount})) 100% repeat-x`;
+
+		// Combine to get the complete CSS background gradient property value
+		return `${tickmarksGradientBase + tickmarksGradientdPattern}`;
+	}
+
+	/**
+	 * Calculates the labels amout, width and text and creates them
+	 *
+	 * @private
+	 */
+	_createLabels() {
+		if (!this.labelInterval || !this.showTickmarks) {
+			return;
+		}
+
+		const labelInterval = this.labelInterval;
+		const step = this._effectiveStep;
+		const newNumberOfLabels = (this._effectiveMax - this._effectiveMin) / (step * labelInterval);
+
+		// If the required labels are already rendered
+		if (newNumberOfLabels === this._oldNumberOfLabels) {
+			return;
+		}
+
+		this._oldNumberOfLabels = newNumberOfLabels;
+		this._labelWidth = 100 / newNumberOfLabels;
+		this._labelValues = [];
+
+		// If the step value is not a round number get its precision
+		const stepPrecision = SliderBase._getDecimalPrecisionOfNumber(step);
+
+		// numberOfLabels below can be float so that the "distance betweenlabels labels"
+		// calculation to be precize (exactly the same as the distance between the tickmarks).
+		// That's ok as the loop stop condition is set to an integer, so it will practically
+		// "floor" the number of labels anyway.
+		for (let i = 0; i <= newNumberOfLabels; i++) {
+			// Format the label numbers with the same decimal precision as the value of the step property
+			const labelItemNumber = ((i * step * labelInterval) + this._effectiveMin).toFixed(stepPrecision);
+			this._labelValues.push(labelItemNumber);
+		}
+	}
+
+	get _labels() {
+		return this._labelValues || [];
+	}
+
+	/**
+	 * Calculates space between tickmarks
+	 *
+	 * @private
+	 */
+	_spaceBetweenTickmarks() {
+		const tickmarksAmountStrCalc = this._tickmarksAmount.split("/");
+		const tickmarksAmount = tickmarksAmountStrCalc[0] / tickmarksAmountStrCalc[1];
+
+		return this.getBoundingClientRect().width / tickmarksAmount;
+	}
+
+	/**
+	 * Notify in case of a invalid step value type
+	 *
+	 * @private
+	 */
+	_validateStep(step) {
+		if (step === 0) {
+			console.warn("The 'step' property must be a positive float number"); // eslint-disable-line
+		}
+
+		if (step < 0) {
+			console.warn("The 'step' property must be a positive float number. The provided negative number has been converted to its positve equivalent"); // eslint-disable-line
+		}
+
+		if (typeof step !== "number" || Number.isNaN(step)) {
+			console.warn("The 'step' property must be a positive float number. It has been set to its default value of 1"); // eslint-disable-line
+		}
+	}
+
+	/**
+	 * Normalizes a new <code>step</code> property value.
+	 * If tickmarks are enabled recreates them according to it.
+	 *
+	 * @private
+	 */
+	get _effectiveStep() {
+		let step = this.step;
+
+		if (step === 0) {
+			return;
```

Comment:
do you mean `return step`? As it is currently undefined if 0

Reply:
Actually this is a mistake of mine, if 0 or a positive number this function is supposed return it. As it is now it doesn't make any difference inthe Slider - everything is working fine, but in the Range Slider it affects the UI. 

I fixed it in the RangeSlider pull request (it is based on this one to use this base class and the shared CSS), when we submit it it will be fixed.